src/hashmap.c

```diff
@@ -69,13 +54,12 @@ void hm_add(Node** hmap, char key[], char value[]){
             strcpy(node->value, value);
             return;
         }
-        if (node->next == NULL){
```

Comment:
Normally you'll null-check a pointer by putting it directly in a conditional statement. It does the same thing either way but not comparing to NULL is more common and also nicer imo

Reply:
Makes sense